lib/tasks/upload_source_strings_for_translation.rake

```diff
@@ -15,42 +15,36 @@ namespace :i18n_sonder do
       klasses = ApplicationRecord.subclasses.select { |m| m.included_modules.include?(Mobility::ActiveRecord) }
     end
 
-    uploader = I18nSonder::Workers::UploadSourceStringsWorker.new
-
     klasses.each do |klass|
-      Rails.logger.info("[upload_source_strings_for_translation] Beginning upload for #{klass}")
+      I18nSonder.logger.info("[upload_source_strings_for_translation] Beginning upload for #{klass}")
       total = 0
 
       klass.in_batches.each do |batch|
         batch.each do |obj|
-          # Check if model has the method defined and if it evaluates to true
-          model_allowed_for_translation = !klass.method_defined?(:allowed_for_translation?) ||
-              obj.allowed_for_translation?
-
-          if model_allowed_for_translation
-            # Get translated attributes, and each attribute's params for uploading translations
-            translated_attribute_names = klass.translated_attribute_names
-            translated_attribute_params = {}
-            translated_attribute_names.each do |attribute_name|
-              upload_options = obj.public_send("#{attribute_name}_backend").options[:upload_for_translation]
-              translated_attribute_params[attribute_name] = upload_options.is_a?(Hash) ? upload_options : {}
-            end
-
-            namespace = obj.namespace_for_translation.compact if klass.method_defined?(:namespace_for_translation)
-
-            uploader.perform(
-                obj.class.name,
-                obj[:id],
-                {
-                  translated_attribute_params: translated_attribute_params,
-                  namespace: namespace
-                }
-            )
-          end
+          I18nSonder::Upload.new(obj).upload(I18n::default_locale)
         end
 
         total += batch.count
-        Rails.logger.info("[upload_source_strings_for_translation] Uploaded source strings for #{total} #{klass} objects")
+        I18nSonder.logger.info("[upload_source_strings_for_translation] Uploaded source strings for #{total} #{klass} objects")
+      end
+    end
+  end
+
+  desc "Upload the source strings for a given model name and id"
+  task :upload_source_strings_for_object, [:model_name, :model_id] => :environment do |_task, args|
+
+    if !args[:model_name].present? || !args[:model_id].present?
+      I18nSonder.logger.error("[upload_source_strings_for_translation] Beginning upload for #{klass}")
```

Comment:
nit: should this log to something other than error? the message seems like it's a positive flow

Reply:
Ahh, it should be an error, the message is wrong.